### ISSUE TITLE
checkpoint issue

### ISSUE THREAD

**Issue by pdcoded** [mozilla/DeepSpeech#534]
Right now I can see in my checkpoint file of checkpoint directory that there are 5 checkpoints .How can I  start testing the model when the validation loss was least . I didnt stop the training process and validation loss kept on increasing. I know the epoch number for that least validation loss step.

2)How to increase number of checkpoints that get saved ? Does it save checkpoint after every checkpoint_sec 

3)If checkpoint is all_model_checkpoint_paths: "model.ckpt-10225" ,what is 10225 ? Is it the number of batch till now ? Can it be changed to epoch since it is more intuitive .

**Comment by tilmankamp** [comment]
@pdcoded 
1) Unfortunately we have no early stopping implementation yet. For now you have to manually observe your training's loss progress. Would be a great contribution. I'll open an issue for this.
2) At the moment there is no direct code/variable to increase the number of saved checkpoints. One could pass a custom [CheckpointSaverHook](https://www.tensorflow.org/api_docs/python/tf/train/CheckpointSaverHook) into the `MonitoredTrainingSession`. Another PR? I'll open an issue for this.
3) This is the number of `global_step`s - explained [here](https://stackoverflow.com/questions/41166681/what-does-tensorflow-global-step-mean). No, this cannot and should not be changed to epochs, as checkpoint creation is totally independent of epochs. TensorFlow treats checkpoints only time-wise or (global)step-wise. We also want to keep things as TensorFlow'ish as possible and try to avoid code piles on top of it.
